### PR TITLE
feat: Config to eventstore

### PR DESCRIPTION
**Pull request by fgerschwiler**

### CODE REVIEW

File: internal/command/command.go [hunk 1]
```diff
@@ -32,17 +31,14 @@ type Commands struct {
...
-	initializeUserCode          crypto.Generator
-	emailVerificationCode       crypto.Generator
-	phoneVerificationCode       crypto.Generator
-	passwordVerificationCode    crypto.Generator
-	passwordlessInitCode        crypto.Generator
 	machineKeyAlg               crypto.EncryptionAlgorithm
 	machineKeySize              int
 	applicationKeySize          int
-	applicationSecretGenerator  crypto.Generator
+	PasswordHashAlg             crypto.HashAlgorithm
```
Comment (livio-a) [inline internal/command/command.go:41]: UserCodeAlg and PasswordHashAlg are never used in commands itself, please declare and initialise them where needed

File: internal/command/setup_step1.go [hunk 2]
```diff
@@ -134,7 +135,10 @@ func (c *Commands) SetupStep1(ctx context.Context, step1 *Step1) error {
 					EmailAddress:    organisation.Owner.Email,
 					IsEmailVerified: true,
 				},
-			}, orgIAMPolicy, pwPolicy, nil, false)
+			}, orgIAMPolicy, pwPolicy,
+			nil, //TODO: Code Generator missing!
+			nil, //TODO: Code Generator missing!
```
Comment (livio-a) [inline internal/command/setup_step1.go:140]: just as notice: for v2 we'll have to set them up in step1 / createIAM / createTenant and pass them along

File: internal/api/grpc/server/middleware/translation_interceptor.go [hunk 3]
```diff
@@ -3,15 +3,14 @@ package middleware
...
+	"github.com/caos/zitadel/internal/query"
 	"google.golang.org/grpc"
 
 	_ "github.com/caos/zitadel/internal/statik"
 )
 
-func TranslationHandler(defaultLanguage language.Tag) func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
-	translator := newZitadelTranslator(defaultLanguage)
+func TranslationHandler(query *query.Queries) func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+	translator := newZitadelTranslator(query.GetDefaultLanguage(context.TODO()))
```
Comment (livio-a) [inline internal/api/grpc/server/middleware/translation_interceptor.go:13]: please replace context.TODO()

**Review by livio-a — CHANGES_REQUESTED** [review]

File: internal/api/grpc/admin/iam_settings.go [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+	if err != nil {
+		return nil, err
+	}
+	return &admin_pb.ListSecretGeneratorsResponse{
+		Details: object.ToListDetails(result.Count, result.Sequence, result.Timestamp),
+	}, nil
+}
+
+func (s *Server) GetSecretGenerator(ctx context.Context, req *admin_pb.GetSecretGeneratorRequest) (*admin_pb.GetSecretGeneratorResponse, error) {
+	generator, err := s.query.SecretGeneratorByType(ctx, req.GetGeneratorType())
```
[the remark this answers is not in the document]
Comment (fgerschwiler) [reply internal/api/grpc/admin/iam_settings.go]: changed to enum, this should solve the problem

File: internal/repository/iam/secret_generator.go [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+	"github.com/caos/zitadel/internal/eventstore"
+	"github.com/caos/zitadel/internal/eventstore/repository"
+)
+
+const (
+	UniqueSecretGeneratorType       = "secret_generator"
+	secretGeneratorPrefix           = "secret.generator."
+	SecretGeneratorAddedEventType   = iamEventTypePrefix + secretGeneratorPrefix + "added"
+	SecretGeneratorChangedEventType = iamEventTypePrefix + secretGeneratorPrefix + "changed"
+	SecretGeneratorRemovedEventType = iamEventTypePrefix + secretGeneratorPrefix + "removed"
```
[the remark this answers is not in the document]
Comment (fgerschwiler) [reply internal/repository/iam/secret_generator.go:19]: from the api this is not accessible, but if we don't need a code anymore we need to be able to remove it somehow

[the remark this answers is not in the document]
Comment (fgerschwiler) [reply internal/repository/iam/secret_generator.go:19]: but i can also remove the whole event?

File: internal/notification/channels/smtp/channel.go [hunk 6]
```diff
@@ -42,7 +46,7 @@ func (email *Email) HandleMessage(message channels.Message) error {
 	if emailMsg.Content == "" || emailMsg.Subject == "" || len(emailMsg.Recipients) == 0 {
 		return caos_errs.ThrowInternalf(nil, "EMAIL-zGemZ", "subject, recipients and content must be set but got subject %s, recipients length %d and content length %d", emailMsg.Subject, len(emailMsg.Recipients), len(emailMsg.Content))
 	}
-
+	emailMsg.SenderEmail = email.fromAddress
```
Comment (livio-a) [inline internal/notification/channels/smtp/channel.go:49]: senderName is not set and i would change `from` fields in Email as well

File: internal/repository/iam/event_default_language.go [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+)
+
+const (
+	DefaultLanguageSetEventType eventstore.EventType = "iam.default.language.set"
+)
+
+type DefaultLanguageSetEvent struct {
+	eventstore.BaseEvent `json:"-"`
+
+	Language language.Tag `json:"defaultLanguage"`
```
Comment (livio-a) [inline internal/repository/iam/event_default_language.go:21]:
```suggestion
	Language language.Tag `json:"language"`
```

File: internal/repository/iam/event_default_language.go [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+	aggregate *eventstore.Aggregate,
+	defaultLanguage language.Tag,
+) *DefaultLanguageSetEvent {
+	return &DefaultLanguageSetEvent{
+		BaseEvent: *eventstore.NewBaseEventForPush(
+			ctx,
+			aggregate,
+			DefaultLanguageSetEventType,
+		),
+		Language: defaultLanguage,
```
Comment (livio-a) [inline internal/repository/iam/event_default_language.go:43]:
```suggestion
	language language.Tag,
) *DefaultLanguageSetEvent {
	return &DefaultLanguageSetEvent{
		BaseEvent: *eventstore.NewBaseEventForPush(
			ctx,
			aggregate,
			DefaultLanguageSetEventType,
		),
		Language: language,
```

**Review by livio-a — APPROVED** [review]

**Pull request merged**